open-craft/ansible-load-balancer
# Initial implemetation

This is an initial implementation for the Ansible role to deploy a load balancing server for the OpenCraft Instance Manager.  The README.md file contains some basic information.
#### Testing instructions
1. Deploy an OpenStack instance in the region GRA1 using the account "OVH - OpenStack - Horizon - OpenCraft IM - Dev".  Use the image "xenial-server-cloudimg-unmodified-20160517".  You probably need to upload an SSH key before deploying the instance.
2. Check out the test playbook in [this branch](https://github.com/open-craft/deployment-deploy-services/tree/smarnach/lb-test).  (First check out the master version of [deployment-secrets-services](https://github.com/open-craft/deployment-secrets-services) if you haven't already; then go the `deploy` subdirectory and check out the `smarnach/lb-test` branch.)
3. In the top directory of the deployment-secrets-services repository, edit the Ansible inventory `hosts and add
   
   ```
   [load-balancer]
   haproxy-test.opencraft.hosting ansible_host=1.2.3.4
   ```
   
   Use the IP address of your newly deployed VM from step 1.
4. From the same directory, run the playbook using
   
   ```
   ansible-playbook -vvv deploy/deploy-lb.yml -u ubuntu --private-key <your-ssh-id> \
       -i hosts -l haproxy-test.opencraft.hosting -e LETSENCRYPT_USE_STAGING=true
   ```

The playbook should run successfully.  Once it's finished, add a backend as desribed in `README.md`.  A simple haproxy backend definition is to forward to some webserver running on localhost, e.g.

```
backend be_pydoc
    server srv_pydoc localhost:8000
```

Then run `pydoc -p 8000` in another terminal.
